include/deal.II/base/tensor.h

```diff
@@ -804,6 +814,15 @@ Tensor<0,dim,Number> &Tensor<0,dim,Number>::operator = (const Tensor<0,dim,Other
 }
 
 
+template <int dim, typename Number>
+inline
+Tensor<0,dim,Number> &Tensor<0,dim,Number>::operator = (const Tensor<0,dim,Number> &p)
+{
+  value = p.value;
```

Comment:
For me using
```
value = internal::NumberType<Number>::value(rhs);
```
instead and not deleting the non-const copy constructor seems to work using ICC-15.

Reply:
I think I finally understand what the problem is here, in conjunction with the error that @QiaoLei-88 posted for the `=default` version: this happens during instantiation with a `Number` type that is poorly implemented and only has a copy constructor that doesn't keep the right hand side `const`. (For example, a type that only has a move constructor, but not a copy constructor.) @masterleinad 's workaround above *explicitly* creates a copy, which can then be moved as necessary.

I don't think that's fundamentally a problem in deal.II, and maybe not even in icc15, but in `Sacado::Rad::ADvar`. The workaround above is probably as good a solution to this as any.